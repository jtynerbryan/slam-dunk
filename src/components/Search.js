import React from 'react';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { searchHighlights } from '../actions/highlights'
import { getHighlights } from '../actions/highlights'
import { updateHighlights } from '../actions/highlights'
import Navbar from './Navbar'
import Loader from './Loader'
import FilteredHighlightCollection from './FilteredHighlightCollection'
import { Button, Form,} from 'semantic-ui-react'

class Search extends React.Component {

  state = {
    searchValue: ''
  }

  componentDidMount() {
    if (!this.props.highlightsUpdated) {
      this.props.updateHighlights()
    }
  }

  componentDidUpdate() {
    if (this.props.highlights.length === 0) {
      this.props.getHighlights()
    }
  }

  handleChange = (e) => {
    this.setState({
      searchValue: e.target.value
    })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    if (this.state.searchValue === "") {
      alert("Please enter something to search")
    } else {
      this.props.searchHighlights(this.state.searchValue)
    }
  }

  render() {
    if (this.props.highlights.length > 0) {
      return (
        <div className="App">
          <h2>Search for highlights</h2>
          <Form onSubmit={this.handleSubmit}>
            <Form.Input placeholder="Most highlight titles reference players, nicknames, or actions (dunk, slam, etc.)" onChange={this.handleChange}></Form.Input>
            <Button type="submit">Submit</Button>
          </Form>
          <br></br>
          <FilteredHighlightCollection/>
        </div>
      )
    } else {
        return (
          <Loader/>
        )
    }

  }

}

function mapStateToProps(state) {
  return {
    highlights: state.highlights,
    filteredHighlights: state.filteredHighlights,
    highlightsUpdated: state.highlightsUpdated
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    searchHighlights,
    getHighlights,
    updateHighlights
  }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Search)
