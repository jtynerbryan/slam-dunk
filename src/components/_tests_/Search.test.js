import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Search } from '../Search';

describe('Search Comonent', () => {
  it('renders without crashing', () => {
    shallow(<Search />);
  });

  it('doesn\'t change unexpectedly', () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
