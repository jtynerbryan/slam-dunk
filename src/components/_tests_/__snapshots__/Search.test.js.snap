// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Comonent doesn't change unexpectedly 1`] = `
<div
  className="App"
>
  <h2>
    Search for highlights
  </h2>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="fields"
    >
      <div
        className="field"
      >
        <div
          className="ui input"
          content={undefined}
        >
          <input
            disabled={undefined}
            onChange={[Function]}
            placeholder="search highlights"
            required={undefined}
            tabIndex={undefined}
            type="text"
          />
        </div>
      </div>
      <button
        className="ui button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
